apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
+	TC_ASSERT_EQ("lib_meminstream", meminstream.buffer, (FAR char *)(buf));
+
+	lib_memsistream((FAR struct lib_memsistream_s *)&memsistream, buf + 2, BUFLEN);
+	TC_ASSERT_EQ("lib_memsistream", (memsistream.buffer - (FAR char *)(buf)), 2);
+	TC_ASSERT_EQ("lib_memsistream", memsistream.buflen, BUFLEN);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_memsostream_tc
+* @brief            Initializes a stream for use with a fixed-size memory buffer
+* @scenario         Initializes a stream for use with a fixed-size memory buffer
+* @apicovered       lib_memsostream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_memsostream_tc(void)
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_memoutstream_s memoutstream;
+	struct lib_memsostream_s memsostream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
+
+	lib_memsostream((FAR struct lib_memsostream_s *)&memsostream, buf + 4, BUFLEN);
+	TC_ASSERT_EQ("lib_memsostream", (memsostream.buffer - (FAR char *)(buf)), 4);
+	TC_ASSERT_EQ("lib_memsostream", memsostream.buflen, (BUFLEN - 1));	/* Save space for null terminator, hence checing with (BUFLEN-1)*/
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_nullinstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will return only EOF
+* @scenario         Initializes a NULL stream. The initialized stream will return only EOF
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+        union {
+                struct lib_instream_s nullinstream;
+                struct lib_meminstream_s meminstream;
+        } instream_object;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_instream_s *stream;
+
+	lib_meminstream((FAR struct lib_meminstream_s *)&instream_object.meminstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_meminstream", instream_object.meminstream.buffer, (FAR char *)(buf));
+
+	lib_nullinstream(&instream_object.nullinstream);
+	stream = &instream_object.nullinstream;
+	TC_ASSERT_EQ("lib_nullinstream", instream_object.nullinstream.nget, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_nulloutstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @scenario         Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @apicovered       lib_nulloutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nulloutstream_tc(void)
+{
+        union {
+                struct lib_outstream_s nulloutstream;
+                struct lib_memoutstream_s memoutstream;
+        } outstream_object;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_outstream_s *stream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&outstream_object.memoutstream, buf, BUFLEN);
```

Comment:
same above. I don't understand these codes are needed for this tc.

Reply:
Yes i crosschecked, i think we dont need it for nullstreams